### ISSUE TITLE
Need support for #bold( )

### ISSUE THREAD

**Issue by ethanpixar** [supersurviveur/typst-math#31]
I am working on Matrix and Vectors, and the code in typst would look like these:
<img width="1157" alt="image" src="https://github.com/user-attachments/assets/0a163506-0162-45c7-8e22-c24f108fb975">
I wonder whether I can add these decrorations myself, so I struggled with the json file and failed. 

Hope you can add support for #bold( ) function to make the code more readable!!

**Comment by supersurviveur** [comment]
Hi !
Implementing bold in special cases is probably possible.
My current system to propagate style might be a problem for more complex situation (like `bold(sqrt(2))`).

We must match on the bold function, and then add a `font-weight` style on children.

**Comment by ethanpixar** [comment]
Yes, I do find cases where the system can not render styles correctly, for example the `dash(a)` will look like `—(a)` instead of `ā`. Currently I just want to add support for `bold()`, and it would improve readability so much even just by replacing all the `bold(a)` with just `a`. I read through the `README.md` file and tried 
```
{
  "name": "mySymbol",
  "symbol": "|some chars|",
  "category": "operator"
}
```
but did not work for me. How can I implement this? Great thanks for your reply!

**Comment by supersurviveur** [comment]
You can't add custom func styles, you can just replace a string, so it needs to be hard coded with other functions.
But instead of `abs` function for example, which just add bars, `bold` modify the style of its children.
I will try to implement it in the general case, and just in some special cases (with just a string for example) if it doesn't work.

Unfortunately I don't have the time to do something "cleaner".

**Comment by supersurviveur** [comment]
Feels free to open another issues for other functions like `dash`, if you would like to see them implemented. You can even just list them, since I only added the ones I use for my own documents.

**Comment by ethanpixar** [comment]
Great thanks!! You have really solved my confusion!
